wfe/wfe.go

```diff
@@ -1618,3 +1620,52 @@ func uniqueLowerNames(names []string) []string {
 	sort.Strings(unique)
 	return unique
 }
+
+func (wfe *WebFrontEndImpl) RevokeCert(
+	ctx context.Context,
+	logEvent *requestEvent,
+	response http.ResponseWriter,
+	request *http.Request) {
+	body, _, prob := wfe.verifyPOST(ctx, logEvent, request, wfe.lookupJWK)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	// revokeCertReq is the ACME certificate information submitted by the client
+	var revokeCertReq struct {
+		Certificate string `json:"certificate"`
+		Reason      *uint  `json:"reason,omitempty"`
+	}
+	err := json.Unmarshal(body, &revokeCertReq)
+	if err != nil {
+		wfe.sendError(
+			acme.MalformedProblem("Error unmarshaling certificate revocation JSON body"), response)
+		return
+	}
+
+	if revokeCertReq.Reason != nil {
+		r := *revokeCertReq.Reason
+		if r == 7 || r > 10 {
```

Comment:
Can you make named constants for these? You can lift them from Boulder: https://github.com/letsencrypt/boulder/blob/e78a7bdb10a4897bd0636f00f63750afb081412d/revocation/reasons.go#L6-L20